yourbench/pipeline/summarization.py

```diff
@@ -0,0 +1,497 @@
+
+Error Handling & Logging:
+-------------------------
+- All errors are logged using `loguru` to `logs/summarization.log`.
+- The stage attempts to proceed with partial data even if some calls fail, never
+  abruptly terminating the pipeline.
+- In debug mode, additional corpus-level metrics are computed and logged.
+
+Important Notes:
+----------------
+- This stage relies on the `run_inference` utility function from yourbench.utils.inference_engine
+  for concurrency, timeouts, and model management.
+- Summaries are extracted from the model's output by parsing <final_summary> XML tags.
+- If no valid summary is found, the pipeline substitutes a fallback string.
+
+See Also:
+---------
+- yourbench.utils.inference_engine for concurrency logic
+- yourbench.utils.dataset_engine for loading/saving dataset
+"""
+
+from concurrent.futures import ThreadPoolExecutor
+from concurrent.futures import TimeoutError as FuturesTimeoutError
+from typing import Any, Dict, List, Optional
+
+import evaluate
+from datasets import Dataset
+from loguru import logger
+
+from yourbench.utils.dataset_engine import save_dataset, smart_load_dataset
+from yourbench.utils.inference_engine import InferenceCall, run_inference
+from yourbench.utils.parsing_engine import extract_content_from_xml_tags
+from yourbench.utils.prompts import SUMMARIZATION_USER_PROMPT
+
+
+# We add a stage-specific log file for summarization
+logger.add("logs/summarization.log", level="DEBUG", rotation="10 MB", enqueue=True)
+
+# === Utility metric loaders (only used in debug mode) ===
+# They are defined inside the run function to avoid overhead unless needed.
+
+
+def _safe_compute_bleu_score(predictions: List[str], references: List[List[str]]) -> float:
+    """
+    Compute BLEU score safely with error handling.
+
+    Args:
+        predictions (List[str]): The candidate summaries or predicted texts.
+        references (List[List[str]]): List of reference texts. Each element is
+            itself a list of strings (as BLEU can handle multiple references
+            for the same prediction).
+
+    Returns:
+        float: The overall BLEU score for the corpus. Returns 0.0 if an error
+        occurs or if inputs are empty.
+    """
+    try:
+        safe_preds = [p if p else "" for p in predictions]
+        safe_refs = []
+        for ref_list in references:
+            if not isinstance(ref_list, list) or not ref_list:
+                safe_refs.append([""])
+            else:
+                safe_refs.append([r if r else "" for r in ref_list])
+
+        if not safe_preds or not safe_refs:
+            logger.warning("Skipping BLEU due to empty inputs.")
+            return 0.0
+
+        bleu_metric = evaluate.load("bleu")
+        bleu_result = bleu_metric.compute(predictions=safe_preds, references=safe_refs)
+        return bleu_result.get("bleu", 0.0)
+    except Exception as e:
+        logger.error("Error computing BLEU score: {}", str(e))
```

Comment:
You could want to add the samples on which you tried to compute things

Reply:
removed all metric calcuations